tests/unit/parts/plugins/test_ubuntu_kernel_plugin.py

```diff
@@ -0,0 +1,985 @@
+            '[ -d "${DTBS}" ] && mv "${DTBS}" ;parts/ubuntu-kernel/install/dtbs',
+            "FIRMWARE=unpacked-linux-firmware/lib/firmware/5.15.0-143",
+            '[ -d "${FIRMWARE}" ] && mv "${FIRMWARE}" ;parts/ubuntu-kernel/install/firmware',
+            "rm -rf ;parts/ubuntu-kernel/install/boot",
+        ]
+    )
+    cmd_list = [x.strip() for x in cmds if x]
+    return cmd_list
+
+
+@dataclasses.dataclass
+class BuildParameters:
+    """Parameters for the build normally provided by snapcraft."""
+
+    def __init__(
+        self,
+        base: str,
+        arch_build_on: str,
+        arch_build_for: str,
+        arch_triplet_build_for: str,
+    ):
+        self.base = base
+        self.arch_build_for = arch_build_for
+        self.arch_build_on = arch_build_on
+        self.arch_triplet_build_for = arch_triplet_build_for
+
+
+@functools.lru_cache
+def get_project_info_parameters() -> list[BuildParameters]:
+    """Generate a ProjectInfo object common to all tests."""
+    return [
+        BuildParameters(
+            base=base,
+            arch_build_on="amd64",
+            arch_build_for=arch_build_for,
+            arch_triplet_build_for=arch_triplet_build_for,
+        )
+        for base in ["core22", "core24"]
+        for arch_build_for in ["amd64", "arm64"]
+        for arch_triplet_build_for in [
+            "x86_64-linux-gnu" if arch_build_for == "amd64" else "aarch64-linux-gnu"
+        ]
+    ]
+
+
+def get_test_fixture_ids() -> list[str]:
+    """Generate fixture ids to help identify tests in logs."""
+    params = get_project_info_parameters()
+    return [f"{x.base}, {x.arch_build_for}" for x in params]
+
+
+@pytest.fixture
+def setup_method_fixture():
+    """Fixture to set up common UbuntuKernelPlugin elements for testing."""
+
+    def _setup_method_fixture(
+        build_params: BuildParameters,
+        new_dir: pathlib.Path,
+        properties: dict[str, str] | None = None,
+    ) -> UbuntuKernelPlugin:
+        if not properties:
+            properties = {"ubuntu-kernel-release-name": "jammy"}
+
+        part = Part("ubuntu-kernel", {})
+
+        project_info = ProjectInfo(
+            application_name="test",
+            cache_dir=new_dir,
+            arch=build_params.arch_build_for,
+            base=build_params.base,
+        )
+
+        part_info = PartInfo(project_info=project_info, part=part)
+        properties_class = UbuntuKernelPlugin.properties_class.unmarshal(properties)
+        return UbuntuKernelPlugin(
+            properties=properties_class,
+            part_info=part_info,
+        )
+
+    yield _setup_method_fixture
+
+
+class TestPluginUbuntuKenrel:
+    """UbuntuKernel plugin tests."""
+
+    def test_property_requires_source_or_release_name(self, new_dir):
+        """Test the property validates source and release name."""
+        properties = {
+            "plugin-name": "ubuntu-kernel",
+        }
+        with pytest.raises(errors.SnapcraftError):
```

Comment:
```suggestion
        with pytest.raises(errors.SnapcraftError, match="<appropriate-error-message-regex>"):
```
To help make sure we're actually testing the behavior we want to test. Same goes for the other `pytest.raises` instances

Reply:
done; I updated the exception message in the plugin as well to replace `_` with `-` so the reported field name matches the yaml field name.